### ISSUE TITLE
[Spell] Mind Flay in raid (BB #108)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#108]
This issue was migrated from bitbucket.
**Original Reporter:** 
**Original Date:** 16.04.2010 20:31:47 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108

<hr>

We had karazhan run on our server, and we noticed that when we have two shadow priests in same grp, we both cannot cast mind flay at the same time on the target. For some bosses it worked.

Example : >  

I start to channel Mind Flay on Phantom Guest.

Then other SP starts to channel Mind Flay too, I will get interrupted msg.

At least you cannot do it while in same grp, dunno about raid.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** niyo
**Original Date:** 16.04.2010 22:03:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156710

<hr>

confirmed, you can't have two mindflay debuffs on one npc

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** sarsikooo
**Original Date:** 16.04.2010 23:15:33 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156723

<hr>

the same warlock's Drain Soul

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** 
**Original Date:** 17.04.2010 07:15:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156828

<hr>

Confirmed the Drain Soul.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** pdx15
**Original Date:** 17.04.2010 09:37:24 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156867

<hr>

http://bitbucket.org/oregon/oregoncore/issue/33/spells-stucks-mind-flay-shadow-weaving-in

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** 
**Original Date:** 17.04.2010 12:01:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156921

<hr>

It's marked as Resolved, that's why I posted this

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** sarsikooo
**Original Date:** 17.04.2010 12:36:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-156925

<hr>

because this is another bug

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** apple
**Original Date:** 20.06.2010 10:03:08 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-199215

<hr>

**Bump!**

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** pdx15
**Original Date:** 20.06.2010 10:17:12 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-199230

<hr>

Duplicate of #33.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** pdx15
**Original Date:** 06.09.2010 13:05:32 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-239642

<hr>

http://bitbucket.org/oregon/oregoncore/changeset/ebc251f5a554 not work

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 24.06.2013 02:57:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4944544

<hr>

The below example is how I resolved it, extremely hacky but it works 100%. **_Changes made in SpellMgr.cpp.**_

```
#!diff
            spellInfo->AttributesEx3 |= SPELL_ATTR_EX3_NO_DONE_BONUS;
            break;
+       case 30708: // Totem of Wrath
+       case 25566: // Healing stream totem
+       case 15407: // Mind flay begin
+       case 17311:
+       case 17312:
+       case 17313:
+       case 17314:
+       case 18807:
+       case 25387: // Mind flay end
+       case 689:   // Drain Life begin
+       case 699:
+       case 709:
+       case 7651:
+       case 11699:
+       case 11700:
+       case 27219:
+       case 27220: // Drain Life end
+           spellInfo->AttributesEx3 |= SPELL_ATTR_EX3_STACK_FOR_DIFF_CASTERS;
+           break;
         case 32727: // Arena Preparation - remove invisibility aura
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 24.06.2013 05:44:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4945311

<hr>

SPELL_ATTR_EX3_STACK_FOR_DIFF_CASTERS - not defined!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 24.06.2013 05:50:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4945347

<hr>

Is this define right?

#define SPELL_ATTR_EX3_STACK_FOR_DIFF_CASTERS     0x00000080            // 7

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 24.06.2013 09:09:46 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4946993

<hr>

I think this fix will not work
as the same as this :
https://bitbucket.org/oregon/oregoncore/issue/357/item-living-ruby-pendant-id-24110

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 24.06.2013 16:17:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4952745

<hr>

It works, I just need to post the changes I made to handle those. A child could figure it out. I'll post it tonight for anyone that can't figure it out

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 25.06.2013 00:59:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4957387

<hr>

Yes, it's all correct PadreWoW. Should work.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 25.06.2013 05:14:39 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4958578

<hr>

![2](http://s018.radikal.ru/i507/1306/a8/a36c17d5a3c0.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 25.06.2013 05:14:58 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-4958581

<hr>

![3](http://s019.radikal.ru/i601/1306/f1/1f0074a1fa3e.jpg)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 30.07.2013 19:10:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5395152

<hr>

It's a hack

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 31.07.2013 05:31:42 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5399597

<hr>

And what?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 31.07.2013 14:04:49 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5405446

<hr>

LOL? and What? gj with 1OO% Hackfixing

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 31.07.2013 15:16:43 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5406528

<hr>

Oh look. It's summer.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 01.08.2013 05:53:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5414018

<hr>

And what? Hackfixing will crush the server or break DBs or scare players or WHAT?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 01.08.2013 13:20:16 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5419146

<hr>

Or Change Sniff/WDB/DBC Data

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.08.2013 13:32:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5432748

<hr>

What change?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 02.08.2013 14:39:19 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5433618

<hr>

(facepalm) man Delete it plz

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.08.2013 16:13:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5434623

<hr>

Another "genius" ......
Post "not hack" fix here or shut you f*king mouth

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 02.08.2013 19:33:16 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5436434

<hr>

It's a group stack, easy to fix. You can try it in DB with minimum support in core rly rly easy just im genius and you are noob in C++ and Developing

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 03.08.2013 09:03:50 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5439961

<hr>

ok ok "genius" i see - you know everything and can fix anything.

p.s it was sarcasm

you're just a talker and noob.

anyone can criticize, but you are not offer anything instead off "hack fix", so SHUT UP

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** saqirmdev
**Original Date:** 03.08.2013 22:07:31 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5443864

<hr>

How long did it take you to write this msg? Pls stop, your comment is useless and ur knowledge is too low.

It's end

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 04.08.2013 10:51:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-5445890

<hr>

Put your "end" into your ASS

This fix works normal and shut up if you can't do another fix!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 04.01.2014 09:22:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-7584824

<hr>

Can someone make pull request with fix from LordUsagi? I was using such a fix during ~1,5 year and got no problems.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Yooki
**Original Date:** 05.01.2014 00:06:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-7590769

<hr>

I will probably get on this tomorrow, although I will try to come up with a different solution if possible because of the complaints made in the posts above.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 07.01.2014 19:24:00 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-7625476

<hr>

Resolved in latest pull request. Please close.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 07.01.2014 21:54:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-7627109

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** release_ok
**Original Date:** 07.01.2014 21:55:01 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/108/#comment-7627121

<hr>

Resolved issue #108 - Spell channeling and auras:

Spells such as Mind Flay and Drain Soul could not have be channeled by more then one player
per single target. If the second player decided to channel the same spell at the same target,
the channeling event for the first player would be canceled/interrupted, and the aura would be refreshed.

It took hours of trial and error to try to figure out what was going on 'AddAura', and after
rewriting the section that handles aura stacking, LordUsagi and myself finally made it work.

There is a small visual bug with aura icons that remains unsolved, and it's uncertain
if it was present there before or was it recent changes in code that triggered this odd behavior.
Other then that everything seems to be working in perfect order!

→ <<cset 663c71197c0c>>
